FileSystem/PrintUtil.js

```diff
@@ -0,0 +1,153 @@
+/**
+ * @fileoverview Helper functions and structures for print API
+ */
+
+/**
+ * Print VbaTab
+ * @param(object) file File structure
+ * @param(object) tab Tab structure
+ */
+function printTab(file, tab) {
+  var filePointer = file.pointer;
+
+  // Get index of start of current line
+  var lineStartIndex = getLineStart(file);
+  // Calculate current line length
+  var lineLength = filePointer - lineStartIndex;
+  var insertSpaceCount;
+
+  if (tab.column == 'NEXT_ZONE') {
+    // If tab is next zone, then we need to insert the
+    // next character on the nearest column with multiple of 14.
+    // Calculate spaces to get to nearest column index which is multiple of 14
+    insertSpaceCount = (14 - (lineLength % 14)) % 14;
+  } else {
+    // Calculate spaces to get to the exact column index
+    insertSpaceCount = tab.column - lineLength - 1;
+  }
+
+  printSpace(file, new Space(insertSpaceCount));
+}
+
+/**
+ * Print VbaDate
+ * @param(object) file File structure
+ * @param(Date) date Date structure
+ */
+function printDate(file, date) {
+  var day = date.getDate();
+  var month = date.getMonth() + 1;
+  var year = date.getFullYear();
+
+  // Prefix digit 0 to ensure double digit day/month
+  if (month < 10) {
+    month = '0' + month;
+  }
+  if (day < 10) {
+    day = '0' + day;
+  }
+
+  // Construct date string as per VBA format
+  var str = day + '-' + month + '-' + year + ' ';
+  stringInsert(file, str);
+}
+
+/**
+ * Print space
+ * @param(object) file File structure
+ * @param(object) space Space structure
+ */
+function printSpace(file, space) {
+  var spaces = '';
```

Comment:
can use this instead?

https://stackoverflow.com/questions/1877475/repeat-character-n-times

Reply:
I had actually used .repeat() first, it worked very well locally. I think it will work in V8 run time. But it didn't work in Rhino version. There is another version using an array. I didn't use it because it didn't look very readable to me. But it works. Should I make a change?
```
var spaces = " ".repeat(space.count); // Doesn't work in rhino
 var spaces = Array(space.count + 1).join(" ");; // Works
````

Reply:
Fixed it to `var spaces = Array(space.count + 1).join(" ");`